packages/library/src/common/class.library.ts

```
@@ -128,21 +145,24 @@ export function Serializable<T extends Constructor>(value: T, { kind, name, addI
 
 /** make a Class not instantiable */
 export function Static<T extends Constructor>(value: T, { kind, name }: ClassDecoratorContext<T>): T | void {
-	name = String(name);
+	const finalName = getClassName(value, name);
 
 	switch (kind) {
 		case 'class':
+			const safeName = finalName || value.name || 'Anonymous';
```

Biome (2.4.16)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)